[imanamirshah] iP

imanamirshah opened the pull request.

imanamirshah committed:
Rename chatbot to Alexis

imanamirshah committed:
feat: Add greeting

imanamirshah committed:
feat: Add echo to Alexis.java

Alexis will echo the command entered by the user.

imanamirshah committed:
feat: Implement exit on command `bye` to Alexis.java

Alexis only echos one message and terminates.

Implementing exit on `bye` allows Alexis to echo any number of
messages. When user types in the command `bye`, Alexis will exit.

imanamirshah committed:
feat: Add ability to store text in Alexis.java

imanamirshah committed:
feat: Add ability to list stored text in Alexis.java

Users are not able to see the stored text as they are not displayed.

Let's implement a feature for Alexis to display the stored text when
requested by users through the command `list`.

imanamirshah committed:
refactor: Refactor tasks to new class Task

imanamirshah committed:
feat: Add TaskList class for task handling

TaskList class encapsulates all task operations to add, mark, unmark and
print tasks.

imanamirshah committed:
style: Modify code to comply with coding standard

imanamirshah committed:
fix: Modify Scanner instantiation and loop for input processing

• Move Scanner to a single instance outside the main loop

imanamirshah committed:
feat: Add TaskType enum

imanamirshah committed:
feat: Add subclasses for different Task types

imanamirshah committed:
refactor: Extract method from addTask() to Task subclasses

• getToDo() abstracted to ToDo class
• getDeadline() abstracted to Deadline class
• getEvent() abstracted to Event class
• Improves readibility of addTask() method

imanamirshah committed:
style: Modify print statement for adding and marking tasks

imanamirshah committed:
feat: Implement task addition for TODO, DEADLINE and EVENT types

imanamirshah committed:
refactor: Abstract printing of welcome message to Console class

• Improves readability of main

imanamirshah committed:
test: Add test input and expected output

imanamirshah committed:
test: Modify test scripts

imanamirshah committed:
refactor: Extract printAddTaskMessage() method from addToTaskList()

liuy1103 commented on src/main/java/Console.java:
```diff
@@ -0,0 +1,28 @@
+            + "What can I do for you?";
+    private final static String GOODBYE = "Bye. Hope to see you again soon!";
+
+    public static void printWelcomeMessage() {
+        System.out.println(NAME);
+        System.out.println(LINEBREAK);
+        System.out.println(GREETING);
+        System.out.println(LINEBREAK);
+    }
+
+    public static void printGoodbyeMessage() {
+        System.out.println(LINEBREAK);
+        System.out.println(GOODBYE);
+        System.out.println(LINEBREAK);
+    }
+}
```
Code is very neat and organized!

imanamirshah replied:
Thank you!

liuy1103 commented on src/main/java/TaskType.java:
```diff
@@ -0,0 +1,10 @@
+public enum TaskType {
```
This class seems a little redundant to me, I think having too many subclasses may sometimes unnecessarily complicate the code.

imanamirshah replied:
I see. But I may expand this to an enum of 'Commands' to allow me to switch between different constants in a more reader friendly way. Still thinking about it but thanks for the input!

liuy1103 commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,84 @@
-| | | | | | | |/ / _ \
-| |_| | |_| |   <  __/
-|____/ \__,_|_|\_\___|
+   ('-.                 ('-.  ) (`-.               .-')    
```
Good job for implementing the text UI testing!

imanamirshah replied:
Thanks!

liuy1103 commented on src/main/java/Alexis.java:
```diff
@@ -0,0 +1,48 @@
+                    String firstWord = line.substring(0, firstSpace);
+                    String description = line.substring(firstSpace);
+                    TaskType taskType = TaskType.getTaskType(firstWord);
+                    switch (taskType) {
+                    case TODO:
+                    default:
+                        tasks.addTask(TaskType.TODO, description);
+                        break;
+                    case DEADLINE:
+                        tasks.addTask(TaskType.DEADLINE, description);
+                        break;
+                    case EVENT:
+                        tasks.addTask(TaskType.EVENT, description);
+                        break;
+                    }
+                }
```
Perhaps can consider creating a private static method for this? :)

imanamirshah replied:
Yep will consider. Have not really finished code quality refactoring haha. Thanks!

liuy1103 commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,14 @@
+todo homework
```
Good job! 💯

imanamirshah replied:
;)

liuy1103 reviewed (commented):
Overall LGTM (Looks good to me), keep up the good work! I think some of your code is a good inspiration for me too 💯 :)

yzhanglp commented on src/main/java/Alexis.java:
```diff
@@ -0,0 +1,48 @@
+                    String firstWord = line.substring(0, firstSpace);
+                    String description = line.substring(firstSpace);
+                    TaskType taskType = TaskType.getTaskType(firstWord);
+                    switch (taskType) {
```
Good to use switch here to increase the readibility

yzhanglp commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+        this.by = by;
+    }
+
+    protected static Deadline getDeadline(String input) {
```
Use a function to achieve get deadline is good

yzhanglp commented on src/main/java/Console.java:
```diff
@@ -0,0 +1,28 @@
+public class Console {
+    private final static String NAME = "   ('-.                 ('-.  ) (`-.               .-')    \n"
```
Good design!

yzhanglp commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+    }
+
+    protected static Deadline getDeadline(String input) {
+        String keyword = "/by";
```
Good to create keyword here for better understanding

yzhanglp reviewed (commented):
Well designed and orginazed, good job!

itsmejr257 commented on src/main/java/TaskList.java:
```diff
@@ -2,19 +2,41 @@
             System.out.println("Nice! I've marked this task as done:");
+            System.out.print("\t");
             printTask(task, -1);
         }
```
Great Job at avoiding deep nesting for your implemented functions in TaskList.

itsmejr257 commented on src/main/java/Alexis.java:
```diff
@@ -2,39 +2,22 @@
 
+        Console.printWelcomeMessage();
+
         while (true) {
```
Would it be better to implement this while loop as a separate function to avoid deep nesting in main() ?

imanamirshah replied:
alright will consider. thanks for the suggestion!

itsmejr257 commented on src/main/java/Alexis.java:
```diff
@@ -40,7 +40,22 @@ public static void main(String[] args) {
+                    String firstWord = line.substring(0, firstSpace);
+                    String description = line.substring(firstSpace);
+                    TaskType taskType = TaskType.getTaskType(firstWord);
+                    switch (taskType) {
```
Maybe you can refactor this while loop to use a switch case instead to avoid deep nesting

itsmejr257 commented on src/main/java/TaskList.java:
```diff
@@ -3,13 +3,13 @@
     }
 
-    public void markTask(String input){
+    public void markTask(String input) {
```
Great Job at using making sure verbs are used for methods/functions

itsmejr257 commented on src/main/java/TaskList.java:
```diff
@@ -19,7 +19,7 @@ public void markTask(String input){
     }
 
-    public void unmarkTask(String input){
+    public void unmarkTask(String input) {
```
Great Job at making sure that there is a sufficient level of detail for the names of your functions

itsmejr257 commented on src/main/java/Alexis.java:
```diff
@@ -36,6 +17,8 @@ public static void main(String[] args) {
                 + "What can I do for you?";
         String goodbye = "Bye. Hope to see you again soon!";
 
+        TaskList tasks = new TaskList();
```
Great Job at clearly distinguishing between single-valued and multi-valued variables

itsmejr257 reviewed (commented):
Looks good. The code mostly follows the coding quality standards well! Great Job.

Some suggestions regarding deep nesting provided.

imanamirshah committed:
feat: Add Command enum

• facilitates switch case based on command
• improves scalability when more commands are added

imanamirshah committed:
refactor: Implement switch statement to replace conditionals

• Abstracted the processing of user inputs to Console class
• Utilised switch statements for improved readability
• Extracted methods within switch statements to improve readability

imanamirshah committed:
feat: Add `IllegalArgumentException`

• catches when a command not found in Command is supplied

imanamirshah committed:
feat: Add `NumberFormatException`

• catches when no number is given for commands `mark` and `unmark`

imanamirshah committed:
Merge branch 'branch-Level-5'

imanamirshah committed:
style: Rename `LINEBREAK` to `LINE_BREAK`

imanamirshah committed:
style: Rename `ERROR` to `COMMAND_ERROR`

• differentiates error message strings

imanamirshah committed:
feat: Add `MissingFieldException`

• catches when description is not provided when adding tasks

imanamirshah committed:
Merge branch 'branch-Level-5'

# Conflicts:
#	src/main/java/Console.java

imanamirshah committed:
refactor: Reorganise files into packages

imanamirshah committed:
fix: Modify path to fix test file

imanamirshah committed:
Merge branch 'branch-A-Packages'

imanamirshah committed:
feat: Add `getTaskType` method to retrieve `TaskType` from `String`

imanamirshah committed:
feat: Add getters for `description` and `isDone` in `Task`

imanamirshah committed:
feat: Add getter for `by` in `Deadline`

imanamirshah committed:
feat: Add getters for `start` and `end` in `Event`

imanamirshah committed:
feat: Add getter for `tasks` in `TaskList`

imanamirshah committed:
feat: Implement save to and read from local disk features

• saves file to local disk upon `save` command
• reads from existing file (if any) upon starting

imanamirshah committed:
feat: Enhance save feature

• file automatically saves upon exit with `bye` command

imanamirshah committed:
chore: Include `Alexis.txt` in `.gitignore`

imanamirshah committed:
feat: Add delete task feature

• delete task implemented using ArrayList

imanamirshah committed:
Merge branch 'branch-Level-6'

imanamirshah committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Alexis/console/Command.java

imanamirshah committed:
chore: Create a new JAR file

imanamirshah committed:
fix: Add `IndexOutOfBoundsException`

• catches when no deadline is given to `deadline` command
• catches when no start and end dates/times is given to `event` command

imanamirshah committed:
fix: Modify code to handle edge cases

• handles cases for delete and mark i for i < 0 or i > `numberOfTasks`

imanamirshah committed:
feat: Add `Parser` class to handle user inputs

imanamirshah committed:
refactor: Rename `Save` class to `Storage`

imanamirshah committed:
refactor: Modify `Task` class to abstract class

imanamirshah committed:
refactor: Rename `Console` class to `Ui` class to handle user
interactions

imanamirshah committed:
refactor: Add constants to replace literals

• `MISSING_DEADLINE_ERROR` in `Deadline` class
• `MISSING_EVENT_START_OR_END_DATE_ERROR` in `Event` class
• `ADD_MESSAGE` in `TaskList` class
• `LIST_UPDATE_MESSAGE` in `TaskList` class

imanamirshah committed:
feat: Add feature to find tasks by searching for a keyword

• returns all task that contain the given keyword

imanamirshah committed:
docs: Add Javadoc comments to all non-trivial classes and methods

okkhoy commented on src/main/java/Alexis/Alexis.java:
```diff
@@ -0,0 +1,24 @@
+package Alexis;
```
should be in lower case

okkhoy commented on src/main/java/Alexis/Alexis.java:
```diff
@@ -0,0 +1,24 @@
+        Scanner in = new Scanner(System.in);
+
+        Console.printWelcomeMessage();
+        Console.processUserInput(tasks, in);
```
Shouldn't process input be the responsibility of a different class rather than Console?
In the iP it seems ok; however, you are violating some design principles (that you will learn soon) here. Avoid this in your tP code.

okkhoy commented on src/main/java/Alexis/console/Console.java:
```diff
@@ -0,0 +1,130 @@
+        System.out.println(LINE_BREAK);
+    }
+
+    public static void processUserInput(TaskList tasks, Scanner in) {
```
all non-trivial methods need header comments

okkhoy commented on src/main/java/Alexis/console/Save.java:
```diff
@@ -0,0 +1,167 @@
+        }
+        scanner.close();
+    }
+}
```
Good SLAPping in this class! 👍

okkhoy reviewed (commented):
In general neat code 💯 
Pls add some code documentation as per the increment

imanamirshah committed:
feat: Add date and time formatter

imanamirshah committed:
Merge pull request #1 from imanamirshah/branch-Level-8

feat: Add date and time formatter

imanamirshah committed:
Merge branch 'master' of https://github.com/imanamirshah/ip

imanamirshah committed:
Merge branch 'master' into branch-Level-9

imanamirshah committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/Alexis/console/Parser.java

imanamirshah committed:
Merge pull request #2 from imanamirshah/branch-Level-9

feat: Add feature to find tasks by searching for a keyword

imanamirshah committed:
Merge branch 'master' of https://github.com/imanamirshah/ip

imanamirshah committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/Alexis/task/TaskList.java

imanamirshah committed:
Merge pull request #3 from imanamirshah/branch-A-JavaDoc

docs: Add Javadoc comments to all non-trivial classes and methods

imanamirshah committed:
Merge branch 'master' of https://github.com/imanamirshah/ip

imanamirshah committed:
refactor: Rename `Alexis` package to `alexis`

imanamirshah committed:
Merge branch 'master' into branch-A-JavaDoc

imanamirshah committed:
docs: Add Javadoc comments to methods for `find` feature

imanamirshah committed:
Merge pull request #4 from imanamirshah/branch-A-JavaDoc

Branch a java doc

imanamirshah committed:
Merge branch 'master' of https://github.com/imanamirshah/ip

imanamirshah committed:
test: Fix test file and modify test inputs

imanamirshah committed:
style: Modify code to follow Java coding standard

imanamirshah committed:
feat: Modify `save` feature to save every time task list changes

imanamirshah committed:
refactor: Shift `processUserInput` method to `Parser` class

imanamirshah committed:
test: Modify expected test output after `save` feature update

imanamirshah committed:
feat: Modify `find` feature to be case-insensitive

imanamirshah committed:
docs: Update README.md

imanamirshah committed:
feat: Add `help` command to display all commands and their uses

imanamirshah committed:
fix: Modify code to reject empty string inputs when adding tasks

imanamirshah committed:
docs: Update README.md to include link to A-Release

imanamirshah committed:
docs: Update README.md for punctuation formatting

imanamirshah committed:
docs: Update README.md to correct punctuation
